Get blocking time formula (#38)
11michalis11/AmbulanceDecisionGame: tex/main/MarkovChain/expressions_from_pi/main.tex

    \end{cases}
\end{equation}
 
In equation \ref{eq:time_in_state_blocking_time}, both service completions and 
\textit{other} patient arrivals are considered. 
Thus, from a blocked patient's perspective whenever the system moves from one state 
to another it can either be because of a service being completed or from an arrival 
of \textit{other} patient. 
The following two expressions give the probabilities of these two scenarios:

\begin{equation*}
    p_s(u,v) = \frac{\min(v,C)\mu}{\lambda^o + \min(v,C)\mu}, \qquad
    p_o(u,v) = \frac{\lambda^o}{\lambda^o + \min(v,C)\mu}
\end{equation*}

Having defined \( c(u,v) \) and \(S_b\) a formula for the blocking time that is
expected to occur at each state can be given by:

\begin{equation}\label{eq:blocking-time-at-each-state}
    b(u,v) = 
    \begin{cases} 
        0, & \textbf{if } (u,v) \notin S_b \\
        c(u,v) + b(u - 1, v), & \textbf{if } v = N = T\\
        c(u,v) + b(u, v-1), & \textbf{if } v = N \neq T \\
        c(u,v) + p_s(u,v) b(u-1, v) + p_o(u,v) b(u, v+1), & \textbf{if } u > 0 \textbf{ and } v = T \\
        c(u,v) + p_s(u,v) b(u, v-1) + p_o(u,v) b(u, v+1), & \textbf{otherwise} \\

>     \end{cases}

Author: Cool. So a Markov model that shows exactly how these expressions are formed?

\end{equation}

Unlike equations (\ref{eq:waiting-time-at-each-state-other}) and 
(\ref{eq:waiting-time-at-each-state-ambulance}), equation 
(\ref{eq:blocking-time-at-each-state}) cannot be solved recursively. 

> An algebraic approach will be used to solve this equation here. 

Reviewer: Yeah algebraic isn't quite right. Arguably doing things recursively is algebraic.

I'm not entirely what the correct term is:

- Direct approach;
- Linear algebraic approach;

Author: I agree, algebraic may not be the right name for it. Should we settle for linear algebraic approach for now? (I think I like it slightly better). There may even be a third name that describes it a better though.

Author: Could even be `Solving the linear system approach`

> By finding all equations \(b(u,v)\) for all states \((u,v)\) that belong in \(S_b\) 

Reviewer: ```suggestion
By enumerating all equations of (\ref{eq:blocking-time-at-each-state}) for all states \((u,v)\) that belong in \(S_b\) 
```

a system of linear equations arises where the unknown variables are all the \(b(u,v)\)
terms.
For instance, let us consider a Markov model where \(C=2, T=3, N=6, M=2\). 

> The Markov model and the equivalent equations are shown below. 

Reviewer: ```suggestion
The Markov model is shown in Figure ... and the equivalent equations are (\ref{eq:first_equation_of_example})-(eq:last_equation_of_example).
```

Avoid referring to things based on their position. Their position can change.

The equations considered here are only the ones that correspond to the blocking 
states.

\begin{multicols*}{2}

>     \scalebox{0.68}{\input{MarkovChain/expressions_from_pi/example_model_2362/main.tex}}

Reviewer: Turn this in to a figure environment.

    \columnbreak
    \vspace*{0cm} 

>     \begin{align*}
>         b(1,2) &= c(1,2) + p_o b(1,3) \\
>         b(1,3) &= c(1,3) + p_s b(1,2) + p_o b(1,4) \\
>         b(1,4) &= c(1,4) + b(1,3) \\
>         b(2,2) &= c(2,2) + p_s b(1,2) + p_o b(2,3) \\
>         b(2,3) &= c(2,3) + p_s b(2,2) + p_o b(1,4) \\
>         b(2,4) &= c(2,4) + b(2,3)
>     \end{align*}

Reviewer: ```suggestion
    \begin{align}
        b(1,2) &= c(1,2) + p_o b(1,3) \label{eq:first_equation_of_example}
\\
        b(1,3) &= c(1,3) + p_s b(1,2) + p_o b(1,4) \\
        b(1,4) &= c(1,4) + b(1,3) \\
        b(2,2) &= c(2,2) + p_s b(1,2) + p_o b(2,3) \\
        b(2,3) &= c(2,3) + p_s b(2,2) + p_o b(1,4) \\
        b(2,4) &= c(2,4) + b(2,3)\label{eq:last_equation_of_example}
    \end{align}
```

\end{multicols*}

> Additionally, the above equations can be transformed into the following linear system.

Reviewer: ```suggestion
Additionally, the above equations can be transformed into a linear system of the form \(Mx=y\) with:
```

Reviewer: Not sure if `M, x, y` are the best ones but something like this.

\begin{equation}\label{eq:example-algebaric-approach-blocking-time}

>     \begin{pmatrix}

Reviewer: ```suggestion
    M=
    \begin{pmatrix}
```

        -1 & p_o & 0 & 0 & 0 & 0 \\ %(1,2)
        p_s & -1 & p_o & 0 & 0 & 0 \\ %(1,3)
        0 & 1 & -1 & 0 & 0 & 0 \\ %(1,4)
        p_s & 0 & 0 & -1 & p_o & 0\\ %(2,2)
        0 & 0 & 0 & p_s & -1 & p_o \\ %(2,3)
        0 & 0 & 0 & 0 & 1 & -1 \\ %(2,4)
    \end{pmatrix}

>     \begin{pmatrix}

Reviewer: ```suggestion
    x=
    \begin{pmatrix}
```

        b(1,2) \\
        b(1,3) \\
        b(1,4) \\
        b(2,2) \\
        b(2,3) \\
        b(2,4) \\
    \end{pmatrix}
    = 

>     \begin{pmatrix}

Reviewer: ```suggestion
    y=
    \begin{pmatrix}
```

        -c(1,2) \\
        -c(1,3) \\
        -c(1,4) \\
        -c(2,2) \\
        -c(2,3) \\
        -c(2,4) \\
    \end{pmatrix}
\end{equation}

A more generalised form of equation \ref{eq:example-algebaric-approach-blocking-time}

> can thus be given for any value of \(C,T,N,M\) by:

Reviewer: Yeah, `M` is for sure the wrong letter for my suggestion for the linear system.

Author: I'm gonna use `Z`. Running out of letters though 😆

Reviewer: `Z` is not idea (it's "calculusy") but yeah go with that, we can always change at a later date :) :+1:

> \begin{equation}\label{eq:general-algebaric-approach-blocking-time}

Reviewer: So before this write out explicitly/rigorously the terms of the linear system and then go: `which will resemble:` followed by the `...` stuff.

(Let me know if that's not clear)

Author: So write down the actual generalised equations before the matrix? (similar to what I did with the example model)

Reviewer: Yup :+1:
